src/dbnode/storage/shard.go

```diff
@@ -1503,30 +1510,45 @@ func (s *dbShard) insertSeriesBatch(inserts []dbShardInsert) error {
 			// this method (insertSeriesBatch) via `entryRefCountIncremented` mechanism.
 			entry.OnIndexPrepare()
 
-			id := entry.Series.ID()
-			tags := entry.Series.Tags().Values()
-
-			var d doc.Document
-			d.ID = id.Bytes() // IDs from shard entries are always set NoFinalize
-			d.Fields = make(doc.Fields, 0, len(tags))
-			for _, tag := range tags {
-				d.Fields = append(d.Fields, doc.Field{
-					Name:  tag.Name.Bytes(),  // Tags from shard entries are always set NoFinalize
-					Value: tag.Value.Bytes(), // Tags from shard entries are always set NoFinalize
-				})
+			// Don't insert cold index writes into the index insert queue.
+			if s.coldWritesEnabled &&
+				writeType == series.ColdWrite &&
+				pendingIndex.timestamp.Before(warmIndexBlockStart) {
+				indexBlockStart := s.reverseIndex.BlockStartForWriteTime(pendingIndex.timestamp)
+				// NB(bodu): We finalize this entry to record an index attempt. This means that
+				// we will not attempt to index this series for this time block again.
+				// entry.NeedsIndexUpdate(indexBlockStart) will evaluate falsy.
```

Comment:
`entry.OnIndexFinalize(indexBlockStart)` will actually just mark that it's not being attempted right now
```go
// OnIndexFinalize marks any attempt for the given block start is finished.
func (entry *Entry) OnIndexFinalize(blockStartNanos xtime.UnixNano) {
	entry.reverseIndex.Lock()
	entry.reverseIndex.setAttemptWithWLock(blockStartNanos, false)
	entry.reverseIndex.Unlock()
	// indicate the index has released held reference for provided write
	entry.DecrementReaderWriterCount()
}
```

So wonder if we actually want to also mark it as successful (so consequent writes do not try to index this block again).

e.g.
```go
entry. OnIndexSuccess(indexBlockStart)
entry.OnIndexFinalize(indexBlockStart)
```

Reply:
Ah, you're right. I read the function name and thought it was setting the attempt as `true`.